### ISSUE TITLE
Incorrect Launch Path for .bat Files

### ISSUE THREAD

**Issue by Dandraghas** [lutris/lutris#5130]
### Bug description

When i try to launch some game that requires .bat to start, lutris tries to launch executable name instead of specifying full path to executable

should
`/home/user/.local/share/lutris/runners/wine/wine-ge-8-22-x86_64/bin/wine cmd /C /path/to/game/start.bat`

currently:
`/home/user/.local/share/lutris/runners/wine/wine-ge-8-22-x86_64/bin/wine cmd /C start.bat`

strange, but removing `or os.path.dirname(windows_executable) == working_dir` in `lutris/util/wine/wine.py:251`. or doing `os.path.abspath` instead of basename on line 253 will fix this.

### How to Reproduce

1. Try to launch game with .bat file

### Expected behavior

Lutris should construct the launch command with the full path to the .bat file

### Log output

```shell
INFO     2023-11-12 23:24:59,889 [application.do_command_line:436]:Starting Lutris 0.5.14
INFO     2023-11-12 23:24:59,890 [startup.get_drivers:57]:Using NVIDIA drivers 545.29.02 for x86_64
INFO     2023-11-12 23:24:59,890 [startup.get_drivers:61]:GPU: NVIDIA GeForce GTX 1050 Ti
INFO     2023-11-12 23:24:59,890 [display.display_gpu_info:78]:GPU: 10DE:1C82 1458:3764 (nvidia drivers)
DEBUG    2023-11-12 23:25:00,366 [xrandr._get_vidmodes:16]:Retrieving video modes from XrandR
DEBUG    2023-11-12 23:25:01,485 [lutris.get_missing_game_ids:202]:Checking for missing games

(lutris:81092): Gtk-WARNING **: 23:25:01.498: Creating a portal monitor failed: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: Portal operation not allowed: Unable to open /proc/81092/root
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/lutris/style_manager.py", line 77, in _call_cb
    values = obj.call_finish(result)
             ^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: Portal operation not allowed: Unable to open /proc/81092/root (9)
DEBUG    2023-11-12 23:25:13,522 [xrandr._get_vidmodes:16]:Retrieving video modes from XrandR

(lutris:81092): Gtk-WARNING **: 23:25:13.571: Calling org.freedesktop.portal.Inhibit.Inhibit failed: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: Portal operation not allowed: Unable to open /proc/81092/root
Started initial process 81510 from gamemoderun /home/dandraghas/.local/share/lutris/runners/wine/wine-ge-8-22-x86_64/bin/wine cmd /C start.bat
Start monitoring process.
gamemodeauto: 
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Application could not be started, or no application associated with the specified file.
ShellExecuteEx failed: File not found.

Monitored process exited.
Initial process has exited (return code: 256)
Exit with return code 256
DEBUG    2023-11-12 23:25:14,958 [command.on_stop:195]:Process 81508 has terminated with code 256
DEBUG    2023-11-12 23:25:15,592 [game.beat:851]:Game thread stopped
WARNING  2023-11-12 23:25:15,592 [game.on_game_quit:888]:Game still running (state: running)
INFO     2023-11-12 23:25:15,592 [game.stop:862]:Stopping Testing (wine)
DEBUG    2023-11-12 23:25:15,593 [game.stop_game:818]:Testing (wine) has run for 2 seconds
WARNING  2023-11-12 23:25:15,593 [game.stop_game:820]:The game has run for a very short time, did it crash?
DEBUG    2023-11-12 23:25:15,593 [application.on_game_stop:721]:Removing 22 from running IDs
DEBUG    2023-11-12 23:25:15,607 [game.stop_game:830]:Playtime: Less than a minute
DEBUG    2023-11-12 23:25:15,607 [game.on_game_quit:906]:Testing stopped at Sun, 12 Nov 2023 23:25:15
DEBUG    2023-11-12 23:25:16,986 [lutris.add_to_path_cache:160]:Adding Testing (wine) to path cache
```

### System Information

```shell
[System]
OS:              Arch Linux rolling n/a
Arch:            x86_64
Kernel:          6.6.1-zen1-1-zen
Desktop:         KDE
Display Server:  x11

[CPU]
Vendor:          GenuineIntel
Model:           Intel(R) Core(TM) i3-10100F CPU @ 3.60GHz
Physical cores:  4
Logical cores:   8

[Memory]
RAM:             15.5 GB
Swap:            12.0 GB

[Graphics]
Vendor:          NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce GTX 1050 Ti/PCIe/SSE2
OpenGL Version:  4.6.0 NVIDIA 545.29.02
OpenGL Core:     4.6.0 NVIDIA 545.29.02
OpenGL ES:       OpenGL ES 3.2 NVIDIA 545.29.02
Vulkan Version:  1.3.269
Vulkan Drivers:  NVIDIA GeForce GTX 1050 Ti (1.3.260)
```

**Comment by danieljohnson2** [comment]
That code you mention does reduce the path to just the file name; it does this if the .bat file is in your working directory. That should be fine; I just tested it, and it worked for me- the .bat file ran as expected.

There's more to this. What's your working directory set to, for this game?

**Comment by Dandraghas** [comment]
> What's your working directory set to, for this game?

it's empty. 
![image](https://github.com/lutris/lutris/assets/91272905/356d4e4c-b8a6-4228-8b14-edc3d0b9e1e4)
setting it to directory of the game still not fix the issue

> That code you mention does reduce the path to just the file name

it doesn't reduce it on .exe files so why it reducing it for .bat?

**Comment by danieljohnson2** [comment]
I wasn't on the project at the time, but I think I can guess. It's to reduce the breakage due to Windows and Linux paths not matching.

With a .bat, the command you run is actually ``cmd "<file.bat>"``, and Wine does not know that the thing in the command line is a file path.

So if your path is ``/home/dangraghas/Game/fun/start.bat``, Windows understands that as ``X:\home\dangraghas\Game\fun\start.bat``, where X is the current drive. Wine starts with Z as current, which makes this work. But if the .bat file selects a different current drive, that'll suddenly break the path. It's absolute to Linux, but drive-relative to Windows.

By handling files in the working directory specially, we work around this. The relative path is just "start.bat", and the working directory is provided to Wine by running it with the working directory set in Linux already- and Wine knows *that* is a Linux path.

If you have no working directory, Lutris picks one for you- it's the one containing your .bat file. Very neat.

You could work around this by picking *some other* directory as the working directory, if that does not break your .bat file.

What I do not understand is why all this is not working for you out of the box. Does your game run if you remove the offending code so the full path is used here?

The thing is, the error is not telling me what file it can't find, but I do not think it is the .bat file. ``cmd /c`` should not be using ShellExecuteEx to run your script, so what is using that API, and what is it trying to execute?

**Comment by ronillon** [comment]
I have just recently came across this issue (start_server.bat not starting)

As @danieljohnson2  points out, the batch files in the working directory have their path stripped to just name.bat.
That means during execution, you can end up with:
`wine cmd /C pause.bat`
The batch file name will be interpreted as a command:
`cmd /C pause`

Example: PAUSE
`Started initial process 154613 from gamemoderun` 
`/home/user/.local/share/lutris/runners/wine/wine-lutris-GE-Proton8-24-x86_64/bin/wine cmd /C pause.bat`
`Start monitoring process.`
`Press any key to continue... ` <- This is the PAUSE command prompt.
`Monitored process exited.`

In mine and OP's case it was START command (https://ss64.com/nt/start.html). 
Which requires additional command to execute (my guess would be that it tried to execute: `.bat` which doesn't exist) 

Interpreted as command:
`start.bat`
`start-test.bat`
`start_test.bat`

NOT Interpreted as command:
`starttest.bat`

You can see it more clearly, when you use a different name, for example: echo, dir, pause, ...
I wonder what kind of mess one could do with carefully crafted filename.

Example: ECHO
`echo_WARNING_The_Name_Of_The_Bat_File_Is_Executed_As_Batch_Command_WARNING.bat `
will be interpreted as:
`echo _WARNING_The_Name_Of_The_Bat_File_Is_Executed_As_Batch_Command_WARNING.bat`
and following will be printed in the log:
`_WARNING_The_Name_Of_The_Bat_File_Is_Executed_As_Batch_Command_WARNING.bat`

TLDR: Do not use batch filenames, that start with a batch command, such as START, PAUSE.
